src/ring_theory/polynomial/cyclotomic/basic.lean

```diff
@@ -423,6 +423,31 @@ begin
            geom_sum_mul, prod_cyclotomic_eq_X_pow_sub_one h]
 end
 
+lemma cyclotomic_dvd_geom_sum_of_dvd (R) [comm_ring R] [is_domain R] {d n : ℕ} (hdn : d ∣ n)
+  (hd : d ≠ 1) : cyclotomic d R ∣ geom_sum X n :=
+begin
+  rcases n.eq_zero_or_pos with rfl | hn,
+  { simp },
+  rw ←prod_cyclotomic_eq_geom_sum hn,
+  apply finset.dvd_prod_of_mem,
+  simp [hd, hdn, hn.ne']
+end
+
+lemma X_pow_sub_one_mul_cyclotomic_dvd_X_pow_sub_one_of_dvd (R) [comm_ring R] {d n : ℕ}
+  (hd : d ∣ n) (hdn : d < n) : (X ^ d - 1) * cyclotomic n R ∣ X ^ n - 1 :=
```

Comment:
You can use `nat.proper_divisors` instead of `(hd : d ∣ n) (hdn : d < n)`. Moreover, can you provide the quotient explicitly? I mean proving `(X ^ d - 1) * (cyclotomic n R )* P =  X ^ n - 1` . Using `polynomial.eq_cyclotomic_iff `  it should be easy, `P` is the product of the cyclotomic polynomials over `nat.divisors n \ nat.divisors d` or something like this.

Reply:
Hmm, we could write this as either:

```lean
(X ^ d - 1) * ∏ x in n.divisors \ d.divisors, cyclotomic n R = X ^ n - 1,
```

or we could write it as:

```lean
(X ^ d - 1) * cyclotomic n R * ∏ x in n.proper_divisors \ d.divisors, cyclotomic n R = X ^ n - 1
```;

do you think there's a preference for either? The original statment obviously follows from the second, whilst with the first it takes a bit of fiddling (and it's part of the monster Wedderburn proof, so don't super want to mess with it!)

Reply:
Can you write both of them? The first one looks more natural, but if you need the second one we can very well have it.

Reply:
i've now done this, and left the "nice" result as a multiplication, and the other as a `dvd`. If we need the exact details, we can just remove the `use` from that first line!